	Reorganize the alphabets/LANG/ directories, rewriting the
	sub-Makefile alphabets/LANG/Makefile.am as Makefile.am
	fragment incuded by alphabets/Makefile.am.

	Note: these include's are resolved by `automake'.

	The idea is to put everything specific for a language into
	the Makefile.am fragment for that language.

	The fragments are (re-)generated by running 'sh mk-lang.sh'
	(in make-rules/alphabets).

	========

	Drop unused files (from TL2008 amd more).

diff -ur -x Makefile.in -x autom4te.cache xindy.orig/configure.ac xindy/configure.ac
--- xindy.orig/configure.ac	2009-12-03 11:59:47.000000000 +0100
+++ xindy/configure.ac	2009-12-18 10:13:35.000000000 +0100
@@ -110,51 +110,6 @@
 		user-commands/Makefile \
 		make-rules/Makefile \
 		make-rules/alphabets/Makefile \
-		make-rules/alphabets/albanian/Makefile \
-		make-rules/alphabets/belarusian/Makefile \
-		make-rules/alphabets/bulgarian/Makefile \
-		make-rules/alphabets/croatian/Makefile \
-		make-rules/alphabets/czech/Makefile \
-		make-rules/alphabets/danish/Makefile \
-		make-rules/alphabets/dutch/Makefile \
-		make-rules/alphabets/english/Makefile \
-		make-rules/alphabets/esperanto/Makefile \
-		make-rules/alphabets/estonian/Makefile \
-		make-rules/alphabets/finnish/Makefile \
-		make-rules/alphabets/french/Makefile \
-		make-rules/alphabets/general/Makefile \
-		make-rules/alphabets/georgian/Makefile \
-		make-rules/alphabets/german/Makefile \
-		make-rules/alphabets/greek/Makefile \
-		make-rules/alphabets/gypsy/Makefile \
-		make-rules/alphabets/hausa/Makefile \
-		make-rules/alphabets/hebrew/Makefile \
-		make-rules/alphabets/hungarian/Makefile \
-		make-rules/alphabets/icelandic/Makefile \
-		make-rules/alphabets/italian/Makefile \
-		make-rules/alphabets/klingon/Makefile \
-		make-rules/alphabets/kurdish/Makefile \
-		make-rules/alphabets/latin/Makefile \
-		make-rules/alphabets/latvian/Makefile \
-		make-rules/alphabets/lithuanian/Makefile \
-		make-rules/alphabets/lower-sorbian/Makefile \
-		make-rules/alphabets/macedonian/Makefile \
-		make-rules/alphabets/mongolian/Makefile \
-		make-rules/alphabets/norwegian/Makefile \
-		make-rules/alphabets/polish/Makefile \
-		make-rules/alphabets/portuguese/Makefile \
-		make-rules/alphabets/romanian/Makefile \
-		make-rules/alphabets/russian/Makefile \
-		make-rules/alphabets/serbian/Makefile \
-		make-rules/alphabets/slovak/Makefile \
-		make-rules/alphabets/slovenian/Makefile \
-		make-rules/alphabets/spanish/Makefile \
-		make-rules/alphabets/swedish/Makefile \
-		make-rules/alphabets/test1/Makefile \
-		make-rules/alphabets/turkish/Makefile \
-		make-rules/alphabets/ukrainian/Makefile \
-		make-rules/alphabets/upper-sorbian/Makefile \
-		make-rules/alphabets/vietnamese/Makefile \
 		make-rules/inputenc/Makefile \
 		make-rules/styles/Makefile \
 		doc/Makefile \
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/Makefile.am xindy/make-rules/alphabets/Makefile.am
--- xindy.orig/make-rules/alphabets/Makefile.am	2009-12-02 20:34:28.000000000 +0100
+++ xindy/make-rules/alphabets/Makefile.am	2010-01-13 13:01:37.000000000 +0100
@@ -21,787 +21,89 @@
 
 langdir	= $(pkglibdir)/modules/lang
 
-DISTCLEANFILES =
+nobase_nodist_lang_DATA =
+dist_noinst_DATA =
 
-albaniandir = $(langdir)/albanian
-
-ALBANIAN = \
-	albanian/latin1-lang.xdy \
-	albanian/latin1-test.xdy \
-	albanian/latin1.xdy \
-	albanian/utf8-lang.xdy \
-	albanian/utf8-test.xdy \
-	albanian/utf8.xdy
-
-nodist_albanian_DATA = $(ALBANIAN)
-
-DISTCLEANFILES += \
-	albanian/utf8.pl
-
-belarusiandir = $(langdir)/belarusian
-
-BELARUSIAN = \
-	belarusian/cp1251-lang.xdy \
-	belarusian/cp1251-test.xdy \
-	belarusian/cp1251.xdy \
-	belarusian/iso88595-lang.xdy \
-	belarusian/iso88595-test.xdy \
-	belarusian/iso88595.xdy \
-	belarusian/isoir111-lang.xdy \
-	belarusian/isoir111-test.xdy \
-	belarusian/isoir111.xdy \
-	belarusian/utf8-lang.xdy \
-	belarusian/utf8-test.xdy \
-	belarusian/utf8.xdy
-
-nodist_belarusian_DATA = $(BELARUSIAN)
-
-DISTCLEANFILES += \
-	belarusian/utf8.pl
-
-bulgariandir = $(langdir)/bulgarian
-
-BULGARIAN = \
-	bulgarian/cp1251-lang.xdy \
-	bulgarian/cp1251-test.xdy \
-	bulgarian/cp1251.xdy \
-	bulgarian/iso88595-lang.xdy \
-	bulgarian/iso88595-test.xdy \
-	bulgarian/iso88595.xdy \
-	bulgarian/koi8-r-lang.xdy \
-	bulgarian/koi8-r-test.xdy \
-	bulgarian/koi8-r.xdy \
-	bulgarian/utf8-lang.xdy \
-	bulgarian/utf8-test.xdy \
-	bulgarian/utf8.xdy
-
-nodist_bulgarian_DATA = $(BULGARIAN)
-
-DISTCLEANFILES += \
-	bulgarian/utf8.pl
-
-croatiandir = $(langdir)/croatian
-
-CROATIAN = \
-	croatian/cp1250-lang.xdy \
-	croatian/cp1250-test.xdy \
-	croatian/cp1250.xdy \
-	croatian/latin2-lang.xdy \
-	croatian/latin2-test.xdy \
-	croatian/latin2.xdy \
-	croatian/utf8-lang.xdy \
-	croatian/utf8-test.xdy \
-	croatian/utf8.xdy
-
-nodist_croatian_DATA = $(CROATIAN)
-
-DISTCLEANFILES += \
-	croatian/utf8.pl
-
-czechdir = $(langdir)/czech
-
-CZECH = \
-	czech/cp1250-lang.xdy \
-	czech/cp1250-test.xdy \
-	czech/cp1250.xdy \
-	czech/latin2-lang.xdy \
-	czech/latin2-test.xdy \
-	czech/latin2.xdy \
-	czech/utf8-lang.xdy \
-	czech/utf8-test.xdy \
-	czech/utf8.xdy
-
-nodist_czech_DATA = $(CZECH)
-
-DISTCLEANFILES += \
-	czech/utf8.pl
-
-danishdir = $(langdir)/danish
-
-DANISH = \
-	danish/cp1252-lang.xdy \
-	danish/cp1252-test.xdy \
-	danish/cp1252.xdy \
-	danish/latin9-lang.xdy \
-	danish/latin9-test.xdy \
-	danish/latin9.xdy \
-	danish/utf8-lang.xdy \
-	danish/utf8-test.xdy \
-	danish/utf8.xdy
-
-nodist_danish_DATA = $(DANISH)
-
-DISTCLEANFILES += \
-	danish/utf8.pl
-
-dutchdir = $(langdir)/dutch
-
-DUTCH = \
-	dutch/ij-as-ij-latin1-lang.xdy \
-	dutch/ij-as-ij-latin1-test.xdy \
-	dutch/ij-as-ij-latin1.xdy \
-	dutch/ij-as-ij-utf8-lang.xdy \
-	dutch/ij-as-ij-utf8-test.xdy \
-	dutch/ij-as-ij-utf8.xdy \
-	dutch/ij-as-y-utf8-lang.xdy \
-	dutch/ij-as-y-utf8-test.xdy \
-	dutch/ij-as-y-utf8.xdy
-
-nodist_dutch_DATA = $(DUTCH)
-
-DISTCLEANFILES += \
-	dutch/ij-as-ij-utf8.pl \
-	dutch/ij-as-y-utf8.pl
-
-englishdir = $(langdir)/english
-
-ENGLISH = \
-	english/cp1252-lang.xdy \
-	english/cp1252-test.xdy \
-	english/cp1252.xdy \
-	english/latin9-lang.xdy  \
-	english/latin9-test.xdy \
-	english/latin9.xdy \
-	english/utf8-lang.xdy \
-	english/utf8-test.xdy \
-	english/utf8.xdy
-
-nodist_english_DATA = $(ENGLISH)
-
-DISTCLEANFILES += \
-	english/utf8.pl
-
-esperantodir = $(langdir)/esperanto
-
-ESPERANTO = \
-	esperanto/latin3-lang.xdy \
-	esperanto/latin3-test.xdy \
-	esperanto/latin3.xdy \
-	esperanto/utf8-lang.xdy \
-	esperanto/utf8-test.xdy \
-	esperanto/utf8.xdy
-
-nodist_esperanto_DATA = $(ESPERANTO)
-
-DISTCLEANFILES += \
-	esperanto/utf8.pl
-
-estoniandir = $(langdir)/estonian
-
-ESTONIAN = \
-	estonian/cp1252-lang.xdy \
-	estonian/cp1252-test.xdy \
-	estonian/cp1252.xdy \
-	estonian/latin9-lang.xdy \
-	estonian/latin9-test.xdy \
-	estonian/latin9.xdy \
-	estonian/utf8-lang.xdy \
-	estonian/utf8-test.xdy \
-	estonian/utf8.xdy
-
-nodist_estonian_DATA = $(ESTONIAN)
-
-DISTCLEANFILES += \
-	estonian/utf8.pl
-
-finnishdir = $(langdir)/finnish
-
-FINNISH = \
-	finnish/cp1252-lang.xdy \
-	finnish/cp1252-test.xdy \
-	finnish/cp1252.xdy \
-	finnish/latin9-lang.xdy \
-	finnish/latin9-test.xdy \
-	finnish/latin9.xdy \
-	finnish/utf8-lang.xdy \
-	finnish/utf8-test.xdy \
-	finnish/utf8.xdy
-
-nodist_finnish_DATA = $(FINNISH)
-
-DISTCLEANFILES += \
-	finnish/utf8.pl
-
-frenchdir = $(langdir)/french
-
-FRENCH = \
-	french/cp1252-lang.xdy \
-	french/cp1252-test.xdy \
-	french/cp1252.xdy \
-	french/latin9-lang.xdy \
-	french/latin9-test.xdy \
-	french/latin9.xdy \
-	french/utf8-lang.xdy \
-	french/utf8-test.xdy \
-	french/utf8.xdy
-
-nodist_french_DATA = $(FRENCH)
-
-DISTCLEANFILES += \
-	french/utf8.pl
-
-generaldir = $(langdir)/general
-
-GENERAL = \
-	general/cp1252-lang.xdy \
-	general/cp1252-test.xdy \
-	general/cp1252.xdy \
-	general/cp850-lang.xdy \
-	general/cp850-test.xdy \
-	general/cp850.xdy \
-	general/latin9-lang.xdy \
-	general/latin9-test.xdy \
-	general/latin9.xdy \
-	general/utf8-lang.xdy \
-	general/utf8-test.xdy \
-	general/utf8.xdy
-
-nodist_general_DATA = $(GENERAL)
-
-DISTCLEANFILES += \
-	general/utf8.pl
-
-georgiandir = $(langdir)/georgian
-
-GEORGIAN = \
-	georgian/utf8-lang.xdy \
-	georgian/utf8-test.xdy \
-	georgian/utf8.xdy
-
-nodist_georgian_DATA = $(GEORGIAN)
-
-DISTCLEANFILES += \
-	georgian/utf8.pl
-
-germandir = $(langdir)/german
-
-GERMAN = \
-	german/braille-utf8-lang.xdy \
-	german/braille-utf8-test.xdy \
-	german/braille-utf8.xdy \
-	german/din5007-latin1-lang.xdy \
-	german/din5007-latin1-test.xdy \
-	german/din5007-latin1.xdy \
-	german/din5007-utf8-lang.xdy \
-	german/din5007-utf8-test.xdy \
-	german/din5007-utf8.xdy \
-	german/duden-latin1-lang.xdy \
-	german/duden-latin1-test.xdy \
-	german/duden-latin1.xdy \
-	german/duden-utf8-lang.xdy \
-	german/duden-utf8-test.xdy \
-	german/duden-utf8.xdy
-
-nodist_german_DATA = $(GERMAN)
-
-DISTCLEANFILES += \
-	german/braille-utf8.pl \
-	german/din5007-utf8.pl \
-	german/duden-utf8.pl
-
-greekdir = $(langdir)/greek
-
-GREEK = \
-	greek/iso88597-lang.xdy \
-	greek/iso88597-test.xdy \
-	greek/iso88597.xdy \
-	greek/polytonic-utf8-lang.xdy \
-	greek/polytonic-utf8-test.xdy \
-	greek/polytonic-utf8.xdy \
-	greek/translit-latin4-lang.xdy \
-	greek/translit-latin4-test.xdy \
-	greek/translit-latin4.xdy \
-	greek/translit-utf8-lang.xdy \
-	greek/translit-utf8-test.xdy \
-	greek/translit-utf8.xdy \
-	greek/utf8-lang.xdy \
-	greek/utf8-test.xdy \
-	greek/utf8.xdy
-
-nodist_greek_DATA = $(GREEK)
-
-DISTCLEANFILES += \
-	greek/polytonic-utf8.pl \
-	greek/translit-utf8.pl \
-	greek/utf8.pl
-
-gypsydir = $(langdir)/gypsy
-
-GYPSY = \
-	gypsy/northrussian-utf8-lang.xdy \
-	gypsy/northrussian-utf8-test.xdy \
-	gypsy/northrussian-utf8.xdy
-
-nodist_gypsy_DATA = $(GYPSY)
-
-DISTCLEANFILES += \
-	gypsy/northrussian-utf8.pl
-
-hausadir = $(langdir)/hausa
-
-HAUSA = \
-	hausa/utf8-lang.xdy \
-	hausa/utf8-test.xdy \
-	hausa/utf8.xdy
-
-nodist_hausa_DATA = $(HAUSA)
-
-DISTCLEANFILES += \
-	hausa/utf8.pl
-
-hebrewdir = $(langdir)/hebrew
-
-HEBREW = \
-	hebrew/iso88598-lang.xdy \
-	hebrew/iso88598-test.xdy \
-	hebrew/iso88598.xdy \
-	hebrew/utf8-lang.xdy \
-	hebrew/utf8-test.xdy \
-	hebrew/utf8.xdy
-
-nodist_hebrew_DATA = $(HEBREW)
-
-DISTCLEANFILES += \
-	hebrew/utf8.pl
-
-hungariandir = $(langdir)/hungarian
-
-HUNGARIAN = \
-	hungarian/cp1250-lang.xdy \
-	hungarian/cp1250-test.xdy \
-	hungarian/cp1250.xdy \
-	hungarian/latin2-lang.xdy \
-	hungarian/latin2-test.xdy \
-	hungarian/latin2.xdy \
-	hungarian/utf8-lang.xdy \
-	hungarian/utf8-test.xdy \
-	hungarian/utf8.xdy
-
-nodist_hungarian_DATA = $(HUNGARIAN)
-
-DISTCLEANFILES += \
-	hungarian/utf8.pl
-
-icelandicdir = $(langdir)/icelandic
-
-ICELANDIC = \
-	icelandic/cp1252-lang.xdy \
-	icelandic/cp1252-test.xdy \
-	icelandic/cp1252.xdy \
-	icelandic/latin9-lang.xdy \
-	icelandic/latin9-test.xdy \
-	icelandic/latin9.xdy \
-	icelandic/utf8-lang.xdy \
-	icelandic/utf8-test.xdy \
-	icelandic/utf8.xdy
-
-nodist_icelandic_DATA = $(ICELANDIC)
-
-DISTCLEANFILES += \
-	icelandic/utf8.pl
-
-italiandir = $(langdir)/italian
-
-ITALIAN = \
-	italian/latin1-lang.xdy \
-	italian/latin1-test.xdy \
-	italian/latin1.xdy \
-	italian/utf8-lang.xdy \
-	italian/utf8-test.xdy \
-	italian/utf8.xdy
-
-nodist_italian_DATA = $(ITALIAN)
-
-DISTCLEANFILES += \
-	italian/utf8.pl
-
-klingondir = $(langdir)/klingon
-
-KLINGON = \
-	klingon/utf8-lang.xdy \
-	klingon/utf8-test.xdy \
-	klingon/utf8.xdy
-
-nodist_klingon_DATA = $(KLINGON)
-
-DISTCLEANFILES += \
-	klingon/utf8.pl
-
-kurdishdir = $(langdir)/kurdish
-
-KURDISH = \
-	kurdish/bedirxan-latin5-lang.xdy \
-	kurdish/bedirxan-latin5-test.xdy \
-	kurdish/bedirxan-latin5.xdy \
-	kurdish/bedirxan-utf8-lang.xdy \
-	kurdish/bedirxan-utf8-test.xdy \
-	kurdish/bedirxan-utf8.xdy \
-	kurdish/turkish-i-latin5-lang.xdy \
-	kurdish/turkish-i-latin5-test.xdy \
-	kurdish/turkish-i-latin5.xdy \
-	kurdish/turkish-i-utf8-lang.xdy \
-	kurdish/turkish-i-utf8-test.xdy \
-	kurdish/turkish-i-utf8.xdy
-
-nodist_kurdish_DATA = $(KURDISH)
-
-DISTCLEANFILES += \
-	kurdish/bedirxan-utf8.pl \
-	kurdish/turkish-i-utf8.pl
-
-latindir = $(langdir)/latin
-
-LATIN = \
-	latin/utf8-lang.xdy \
-	latin/utf8-test.xdy \
-	latin/utf8.xdy
-
-nodist_latin_DATA = $(LATIN)
-
-DISTCLEANFILES += \
-	latin/utf8.pl
-
-latviandir = $(langdir)/latvian
-
-LATVIAN = \
-	latvian/latin4-lang.xdy \
-	latvian/latin4-test.xdy \
-	latvian/latin4.xdy \
-	latvian/utf8-lang.xdy \
-	latvian/utf8-test.xdy \
-	latvian/utf8.xdy
-
-nodist_latvian_DATA = $(LATVIAN)
-
-DISTCLEANFILES += \
-	latvian/utf8.pl
-
-lithuaniandir = $(langdir)/lithuanian
-
-LITHUANIAN = \
-	lithuanian/latin4-lang.xdy \
-	lithuanian/latin4-test.xdy \
-	lithuanian/latin4.xdy \
-	lithuanian/utf8-lang.xdy \
-	lithuanian/utf8-test.xdy \
-	lithuanian/utf8.xdy
-
-nodist_lithuanian_DATA = $(LITHUANIAN)
-
-DISTCLEANFILES += \
-	lithuanian/utf8.pl
-
-lower_sorbiandir = $(langdir)/lower-sorbian
-
-LOWER_SORBIAN = \
-	lower-sorbian/cp1250-lang.xdy \
-	lower-sorbian/cp1250-test.xdy \
-	lower-sorbian/cp1250.xdy \
-	lower-sorbian/latin2-lang.xdy \
-	lower-sorbian/latin2-test.xdy \
-	lower-sorbian/latin2.xdy \
-	lower-sorbian/utf8-lang.xdy \
-	lower-sorbian/utf8-test.xdy \
-	lower-sorbian/utf8.xdy
-
-nodist_lower_sorbian_DATA = $(LOWER_SORBIAN)
-
-DISTCLEANFILES += \
-	lower-sorbian/utf8.pl
-
-macedoniandir = $(langdir)/macedonian
-
-MACEDONIAN = \
-	macedonian/cp1251-lang.xdy \
-	macedonian/cp1251-test.xdy \
-	macedonian/cp1251.xdy \
-	macedonian/iso88595-lang.xdy \
-	macedonian/iso88595-test.xdy \
-	macedonian/iso88595.xdy \
-	macedonian/isoir111-lang.xdy \
-	macedonian/isoir111-test.xdy \
-	macedonian/isoir111.xdy \
-	macedonian/utf8-lang.xdy \
-	macedonian/utf8-test.xdy \
-	macedonian/utf8.xdy
-
-nodist_macedonian_DATA = $(MACEDONIAN)
-
-DISTCLEANFILES += \
-	macedonian/utf8.pl
-
-mongoliandir = $(langdir)/mongolian
-
-MONGOLIAN = \
-	mongolian/cyrillic-utf8-lang.xdy \
-	mongolian/cyrillic-utf8-test.xdy \
-	mongolian/cyrillic-utf8.xdy
-
-nodist_mongolian_DATA = $(MONGOLIAN)
-
-DISTCLEANFILES += \
-	mongolian/cyrillic-utf8.pl
-
-norwegiandir = $(langdir)/norwegian
-
-NORWEGIAN = \
-	norwegian/latin1-lang.xdy \
-	norwegian/latin1-test.xdy \
-	norwegian/latin1.xdy \
-	norwegian/utf8-lang.xdy \
-	norwegian/utf8-test.xdy \
-	norwegian/utf8.xdy
-
-nodist_norwegian_DATA = $(NORWEGIAN)
-
-DISTCLEANFILES += \
-	norwegian/utf8.pl
-
-polishdir = $(langdir)/polish
-
-POLISH = \
-	polish/cp1250-lang.xdy \
-	polish/cp1250-test.xdy \
-	polish/cp1250.xdy \
-	polish/latin2-lang.xdy \
-	polish/latin2-test.xdy \
-	polish/latin2.xdy \
-	polish/utf8-lang.xdy \
-	polish/utf8-test.xdy \
-	polish/utf8.xdy
-
-nodist_polish_DATA = $(POLISH)
-
-DISTCLEANFILES += \
-	polish/utf8.pl
-
-portuguesedir = $(langdir)/portuguese
-
-PORTUGUESE = \
-	portuguese/latin1-lang.xdy \
-	portuguese/latin1-test.xdy \
-	portuguese/latin1.xdy \
-	portuguese/utf8-lang.xdy \
-	portuguese/utf8-test.xdy \
-	portuguese/utf8.xdy
-
-nodist_portuguese_DATA = $(PORTUGUESE)
-
-DISTCLEANFILES += \
-	portuguese/utf8.pl
-
-romaniandir = $(langdir)/romanian
-
-ROMANIAN = \
-	romanian/latin2-lang.xdy \
-	romanian/latin2-test.xdy \
-	romanian/latin2.xdy \
-	romanian/utf8-lang.xdy \
-	romanian/utf8-test.xdy \
-	romanian/utf8.xdy
-
-nodist_romanian_DATA = $(ROMANIAN)
-
-DISTCLEANFILES += \
-	romanian/utf8.pl
-
-russiandir = $(langdir)/russian
-
-RUSSIAN = \
-	russian/cp1251-lang.xdy \
-	russian/cp1251-test.xdy \
-	russian/cp1251.xdy \
-	russian/iso88595-lang.xdy \
-	russian/iso88595-test.xdy \
-	russian/iso88595.xdy \
-	russian/koi8-r-lang.xdy \
-	russian/koi8-r-test.xdy \
-	russian/koi8-r.xdy \
-	russian/translit-iso-utf8-lang.xdy \
-	russian/translit-iso-utf8-test.xdy \
-	russian/translit-iso-utf8.xdy \
-	russian/utf8-lang.xdy \
-	russian/utf8-test.xdy \
-	russian/utf8.xdy
-
-nodist_russian_DATA = $(RUSSIAN)
-
-DISTCLEANFILES += \
-	russian/translit-iso-utf8.pl \
-	russian/utf8.pl
-
-serbiandir = $(langdir)/serbian
-
-SERBIAN = \
-	serbian/cp1251-lang.xdy \
-	serbian/cp1251-test.xdy \
-	serbian/cp1251.xdy \
-	serbian/iso88595-lang.xdy \
-	serbian/iso88595-test.xdy \
-	serbian/iso88595.xdy \
-	serbian/isoir111-lang.xdy \
-	serbian/isoir111-test.xdy \
-	serbian/isoir111.xdy \
-	serbian/utf8-lang.xdy \
-	serbian/utf8-test.xdy \
-	serbian/utf8.xdy
-
-nodist_serbian_DATA = $(SERBIAN)
-
-DISTCLEANFILES += \
-	serbian/utf8.pl
-
-slovakdir = $(langdir)/slovak
-
-SLOVAK = \
-	slovak/large-cp1250-lang.xdy \
-	slovak/large-cp1250-test.xdy \
-	slovak/large-cp1250.xdy \
-	slovak/large-latin2-lang.xdy \
-	slovak/large-latin2-test.xdy \
-	slovak/large-latin2.xdy \
-	slovak/large-utf8-lang.xdy \
-	slovak/large-utf8-test.xdy \
-	slovak/large-utf8.xdy \
-	slovak/small-cp1250-lang.xdy \
-	slovak/small-cp1250-test.xdy \
-	slovak/small-cp1250.xdy \
-	slovak/small-latin2-lang.xdy \
-	slovak/small-latin2-test.xdy \
-	slovak/small-latin2.xdy \
-	slovak/small-utf8-lang.xdy \
-	slovak/small-utf8-test.xdy \
-	slovak/small-utf8.xdy
-
-nodist_slovak_DATA = $(SLOVAK)
-
-DISTCLEANFILES += \
-	slovak/large-utf8.pl \
-	slovak/small-utf8.pl
-
-sloveniandir = $(langdir)/slovenian
-
-SLOVENIAN = \
-	slovenian/cp1250-lang.xdy \
-	slovenian/cp1250-test.xdy \
-	slovenian/cp1250.xdy \
-	slovenian/latin2-lang.xdy \
-	slovenian/latin2-test.xdy \
-	slovenian/latin2.xdy \
-	slovenian/utf8-lang.xdy \
-	slovenian/utf8-test.xdy \
-	slovenian/utf8.xdy
-
-nodist_slovenian_DATA = $(SLOVENIAN)
-
-DISTCLEANFILES += \
-	slovenian/utf8.pl
-
-spanishdir = $(langdir)/spanish
-
-SPANISH = \
-	spanish/modern-latin1-lang.xdy \
-	spanish/modern-latin1-test.xdy \
-	spanish/modern-latin1.xdy \
-	spanish/modern-utf8-lang.xdy \
-	spanish/modern-utf8-test.xdy \
-	spanish/modern-utf8.xdy \
-	spanish/traditional-latin1-lang.xdy \
-	spanish/traditional-latin1-test.xdy \
-	spanish/traditional-latin1.xdy \
-	spanish/traditional-utf8-lang.xdy \
-	spanish/traditional-utf8-test.xdy \
-	spanish/traditional-utf8.xdy
-
-nodist_spanish_DATA = $(SPANISH)
-
-DISTCLEANFILES += \
-	spanish/modern-utf8.pl \
-	spanish/traditional-utf8.pl
-
-swedishdir = $(langdir)/swedish
-
-SWEDISH = \
-	swedish/cp1252-lang.xdy \
-	swedish/cp1252-test.xdy \
-	swedish/cp1252.xdy \
-	swedish/latin9-lang.xdy \
-	swedish/latin9-test.xdy \
-	swedish/latin9.xdy \
-	swedish/utf8-lang.xdy \
-	swedish/utf8-test.xdy \
-	swedish/utf8.xdy
-
-nodist_swedish_DATA = $(SWEDISH)
-
-DISTCLEANFILES += \
-	swedish/utf8.pl
-
-turkishdir = $(langdir)/turkish
-
-TURKISH = \
-	turkish/latin3-lang.xdy \
-	turkish/latin3-test.xdy \
-	turkish/latin3.xdy \
-	turkish/latin5-lang.xdy \
-	turkish/latin5-test.xdy \
-	turkish/latin5.xdy \
-	turkish/utf8-lang.xdy \
-	turkish/utf8-test.xdy \
-	turkish/utf8.xdy
-
-nodist_turkish_DATA = $(TURKISH)
-
-DISTCLEANFILES += \
-	turkish/utf8.pl
-
-ukrainiandir = $(langdir)/ukrainian
-
-UKRAINIAN = \
-	ukrainian/cp1251-lang.xdy \
-	ukrainian/cp1251-test.xdy \
-	ukrainian/cp1251.xdy \
-	ukrainian/koi8-u-lang.xdy \
-	ukrainian/koi8-u-test.xdy \
-	ukrainian/koi8-u.xdy \
-	ukrainian/utf8-lang.xdy \
-	ukrainian/utf8-test.xdy \
-	ukrainian/utf8.xdy
-
-nodist_ukrainian_DATA = $(UKRAINIAN)
-
-DISTCLEANFILES += \
-	ukrainian/utf8.pl
-
-upper_sorbiandir = $(langdir)/upper-sorbian
-
-UPPER_SORBIAN = \
-	upper-sorbian/cp1250-lang.xdy \
-	upper-sorbian/cp1250-test.xdy \
-	upper-sorbian/cp1250.xdy \
-	upper-sorbian/latin2-lang.xdy \
-	upper-sorbian/latin2-test.xdy \
-	upper-sorbian/latin2.xdy \
-	upper-sorbian/utf8-lang.xdy \
-	upper-sorbian/utf8-test.xdy \
-	upper-sorbian/utf8.xdy
-
-nodist_upper_sorbian_DATA = $(UPPER_SORBIAN)
-
-DISTCLEANFILES += \
-	upper-sorbian/utf8.pl
-
-vietnamesedir = $(langdir)/vietnamese
-
-VIETNAMESE = \
-	vietnamese/utf8-lang.xdy \
-	vietnamese/utf8-test.xdy \
-	vietnamese/utf8.xdy
-
-nodist_vietnamese_DATA = $(VIETNAMESE)
-
-DISTCLEANFILES += \
-	vietnamese/utf8.pl
+DOCS_CP1250 =
+DOCS_CP1252 =
+DOCS_CP850 =
+DOCS_LATIN1 =
+DOCS_LATIN2 =
+DOCS_LATIN3 =
+DOCS_LATIN4 =
+DOCS_LATIN5 =
+DOCS_LATIN9 =
+
+DOCS_CYR_1251 =
+DOCS_CYR_ISO88595 = 
+DOCS_CYR_ISOIR111 =
+DOCS_CYR_KOI8_R =
+DOCS_CYR_KOI8_U =
+
+DOCS_ISO88597 =
+DOCS_ISO88598 =
+
+DOCS_UTF8_LAT =
+DOCS_UTF8_CYR =
+DOCS_UTF8_OTHER =
+
+EXTRA_DIST = mk-lang.sh lang.defs
+CLEANFILES =
+
+subdirs =
+
+## Include all Makefile fragments.
+include albanian/Makefile.am
+include belarusian/Makefile.am
+include bulgarian/Makefile.am
+include croatian/Makefile.am
+include czech/Makefile.am
+include danish/Makefile.am
+include dutch/Makefile.am
+include english/Makefile.am
+include esperanto/Makefile.am
+include estonian/Makefile.am
+include finnish/Makefile.am
+include french/Makefile.am
+include general/Makefile.am
+include georgian/Makefile.am
+include german/Makefile.am
+include greek/Makefile.am
+include gypsy/Makefile.am
+include hausa/Makefile.am
+include hebrew/Makefile.am
+include hungarian/Makefile.am
+include icelandic/Makefile.am
+include italian/Makefile.am
+include klingon/Makefile.am
+include kurdish/Makefile.am
+include latin/Makefile.am
+include latvian/Makefile.am
+include lithuanian/Makefile.am
+include lower-sorbian/Makefile.am
+include macedonian/Makefile.am
+include mongolian/Makefile.am
+include norwegian/Makefile.am
+include polish/Makefile.am
+include portuguese/Makefile.am
+include romanian/Makefile.am
+include russian/Makefile.am
+include serbian/Makefile.am
+include slovak/Makefile.am
+include slovenian/Makefile.am
+include spanish/Makefile.am
+include swedish/Makefile.am
+include test1/Makefile.am
+include turkish/Makefile.am
+include ukrainian/Makefile.am
+include upper-sorbian/Makefile.am
+include vietnamese/Makefile.am
+
+## Make sure all language subdirectories exist.
+BUILT_SOURCES = subdirs
+subdirs:
+	@for d in $(subdirs); do $(MKDIR_P) $$d; done
+.PHONY: subdirs
 
 %utf8.pl : %utf8.pl.in
 	sed -e 's/@.{\([^}]*\)}/\1/g;' <$< >$@
@@ -814,13 +116,6 @@
 	sed -e 's/@.{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)CP850 >$@
 
-romanian/latin2.pl : romanian/utf8.pl
-	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
-	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
-
-DISTCLEANFILES += \
-	romanian/latin2.pl
-
 %latin2.pl : %utf8.pl.in
 	sed -e 's/@u{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
@@ -853,13 +148,6 @@
 	sed -e 's/@u{\([^}]*\)}//g;s/@9{\([^}]*\)}/\1/g;' <$< | \
 	$(CONVERT_FROM_UTF8)ISO-8859-15 >$@
 
-romanian/cp1250.pl : romanian/utf8.pl
-	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
-	$(CONVERT_FROM_UTF8)CP1250 >$@
-
-DISTCLEANFILES += \
-	romanian/cp1250.pl
-
 %cp1250.pl : %utf8.pl.in
 	sed -e 's/@u{\([^}]*\)}//g;' <$< | \
 	$(CONVERT_FROM_UTF8)CP1250 >$@
@@ -911,203 +199,11 @@
 	  fi; \
 	fi
 
-SUBDIRS = \
-	albanian \
-	belarusian \
-	bulgarian \
-	croatian \
-	czech \
-	danish \
-	dutch \
-	english \
-	esperanto \
-	estonian \
-	finnish \
-	french \
-	general \
-	georgian \
-	german \
-	greek \
-	gypsy \
-	hausa \
-	hebrew \
-	hungarian \
-	icelandic \
-	italian \
-	klingon \
-	kurdish \
-	latin \
-	latvian \
-	lithuanian \
-	lower-sorbian \
-	macedonian \
-	mongolian \
-	norwegian \
-	polish \
-	portuguese \
-	romanian \
-	russian \
-	serbian \
-	slovak \
-	slovenian \
-	spanish \
-	swedish \
-	test1 \
-	turkish \
-	ukrainian \
-	upper-sorbian \
-	vietnamese
-
 SCRS = \
 	alphabets-doc.tex \
 	make-rules.pl \
 	utf8-to-tex.sed
 
-ALPHABETS = \
-	$(ALBANIAN) \
-	$(BELARUSIAN) \
-	$(BULGARIAN) \
-	$(CROATIAN) \
-	$(CZECH) \
-	$(DANISH) \
-	$(DUTCH) \
-	$(ENGLISH) \
-	$(ESPERANTO) \
-	$(ESTONIAN) \
-	$(FINNISH) \
-	$(FRENCH) \
-	$(GENERAL) \
-	$(GEORGIAN) \
-	$(GERMAN) \
-	$(GREEK) \
-	$(GYPSY) \
-	$(HAUSA) \
-	$(HEBREW) \
-	$(HUNGARIAN) \
-	$(ICELANDIC) \
-	$(ITALIAN) \
-	$(KLINGON) \
-	$(KURDISH) \
-	$(LATIN) \
-	$(LATVIAN) \
-	$(LITHUANIAN) \
-	$(LOWER_SORBIAN) \
-	$(MACEDONIAN) \
-	$(MONGOLIAN) \
-	$(NORWEGIAN) \
-	$(POLISH) \
-	$(PORTUGUESE) \
-	$(ROMANIAN) \
-	$(RUSSIAN) \
-	$(SERBIAN) \
-	$(SLOVAK) \
-	$(SLOVENIAN) \
-	$(SPANISH) \
-	$(SWEDISH) \
-	$(TURKISH) \
-	$(UKRAINIAN) \
-	$(UPPER_SORBIAN) \
-	$(VIETNAMESE)
-
-DOCS_CP1250 = \
-	croatian/cp1250-doc.tex \
-	czech/cp1250-doc.tex \
-	hungarian/cp1250-doc.tex \
-	lower-sorbian/cp1250-doc.tex \
-	polish/cp1250-doc.tex \
-	slovak/large-cp1250-doc.tex \
-	slovak/small-cp1250-doc.tex \
-	slovenian/cp1250-doc.tex \
-	upper-sorbian/cp1250-doc.tex
-
-DOCS_CP1252 = \
-	danish/cp1252-doc.tex \
-	english/cp1252-doc.tex \
-	estonian/cp1252-doc.tex \
-	finnish/cp1252-doc.tex \
-	french/cp1252-doc.tex \
-	general/cp1252-doc.tex \
-	icelandic/cp1252-doc.tex \
-	swedish/cp1252-doc.tex
-
-DOCS_CP850 = \
-	general/cp850-doc.tex
-
-DOCS_LATIN1 = \
-	albanian/latin1-doc.tex \
-	dutch/ij-as-ij-latin1-doc.tex \
-	german/din5007-latin1-doc.tex \
-	german/duden-latin1-doc.tex \
-	italian/latin1-doc.tex \
-	norwegian/latin1-doc.tex \
-	portuguese/latin1-doc.tex \
-	spanish/modern-latin1-doc.tex \
-	spanish/traditional-latin1-doc.tex
-
-
-DOCS_LATIN2 = \
-	croatian/latin2-doc.tex \
-	czech/latin2-doc.tex \
-	hungarian/latin2-doc.tex \
-	lower-sorbian/latin2-doc.tex \
-	polish/latin2-doc.tex \
-	romanian/latin2-doc.tex \
-	slovak/small-latin2-doc.tex \
-	slovak/large-latin2-doc.tex \
-	slovenian/latin2-doc.tex \
-	upper-sorbian/latin2-doc.tex
-
-DOCS_LATIN3 = \
-	esperanto/latin3-doc.tex \
-	turkish/latin3-doc.tex
-
-DOCS_LATIN4 = \
-	greek/translit-latin4-doc.tex \
-	latvian/latin4-doc.tex \
-	lithuanian/latin4-doc.tex
-
-DOCS_LATIN5 = \
-	kurdish/bedirxan-latin5-doc.tex \
-	kurdish/turkish-i-latin5-doc.tex \
-	turkish/latin5-doc.tex
-
-DOCS_LATIN9 = \
-	danish/latin9-doc.tex \
-	english/latin9-doc.tex \
-	estonian/latin9-doc.tex \
-	finnish/latin9-doc.tex \
-	french/latin9-doc.tex \
-	general/latin9-doc.tex \
-	icelandic/latin9-doc.tex \
-	swedish/latin9-doc.tex
-
-DOCS_CYR_1251 = \
-	belarusian/cp1251-doc.tex \
-	bulgarian/cp1251-doc.tex \
-	macedonian/cp1251-doc.tex \
-	russian/cp1251-doc.tex \
-	serbian/cp1251-doc.tex \
-	ukrainian/cp1251-doc.tex
-
-DOCS_CYR_ISO88595 = \
-	belarusian/iso88595-doc.tex \
-	bulgarian/iso88595-doc.tex \
-	macedonian/iso88595-doc.tex \
-	russian/iso88595-doc.tex \
-	serbian/iso88595-doc.tex
-
-DOCS_CYR_ISOIR111 = \
-	belarusian/isoir111-doc.tex \
-	macedonian/isoir111-doc.tex \
-	serbian/isoir111-doc.tex
-
-DOCS_CYR_KOI8_R = \
-	bulgarian/koi8-r-doc.tex \
-	russian/koi8-r-doc.tex
-
-DOCS_CYR_KOI8_U = \
-	ukrainian/koi8-u-doc.tex
-
 DOCS_CYR = \
 	$(DOCS_CYR_1251) \
 	$(DOCS_CYR_ISO88595) \
@@ -1115,72 +211,6 @@
 	$(DOCS_CYR_KOI8_R) \
 	$(DOCS_CYR_KOI8_U)
 
-
-DOCS_OTHER = \
-	greek/iso88597-doc.tex \
-	hebrew/iso88598-doc.tex
-
-
-DOCS_UTF8_LAT = \
-	albanian/utf8-doc.tex \
-	croatian/utf8-doc.tex \
-	czech/utf8-doc.tex \
-	danish/utf8-doc.tex \
-	dutch/ij-as-ij-utf8-doc.tex \
-	dutch/ij-as-y-utf8-doc.tex \
-	english/utf8-doc.tex \
-	esperanto/utf8-doc.tex \
-	estonian/utf8-doc.tex \
-	finnish/utf8-doc.tex \
-	french/utf8-doc.tex \
-	general/utf8-doc.tex \
-	german/din5007-utf8-doc.tex \
-	german/duden-utf8-doc.tex \
-	greek/translit-utf8-doc.tex \
-	greek/utf8-doc.tex \
-	gypsy/northrussian-utf8-doc.tex \
-	hausa/utf8-doc.tex \
-	hungarian/utf8-doc.tex \
-	icelandic/utf8-doc.tex \
-	italian/utf8-doc.tex \
-	kurdish/bedirxan-utf8-doc.tex \
-	kurdish/turkish-i-utf8-doc.tex \
-	latin/utf8-doc.tex \
-	latvian/utf8-doc.tex \
-	lithuanian/utf8-doc.tex \
-	lower-sorbian/utf8-doc.tex \
-	norwegian/utf8-doc.tex \
-	polish/utf8-doc.tex \
-	portuguese/utf8-doc.tex \
-	romanian/utf8-doc.tex \
-	russian/translit-iso-utf8-doc.tex \
-	slovak/small-utf8-doc.tex \
-	slovak/large-utf8-doc.tex \
-	slovenian/utf8-doc.tex \
-	spanish/modern-utf8-doc.tex \
-	spanish/traditional-utf8-doc.tex \
-	swedish/utf8-doc.tex \
-	turkish/utf8-doc.tex \
-	upper-sorbian/utf8-doc.tex \
-	vietnamese/utf8-doc.tex
-
-DOCS_UTF8_CYR = \
-	belarusian/utf8-doc.tex \
-	bulgarian/utf8-doc.tex \
-	macedonian/utf8-doc.tex \
-	mongolian/cyrillic-utf8-doc.tex \
-	russian/utf8-doc.tex \
-	serbian/utf8-doc.tex \
-	ukrainian/utf8-doc.tex
-
-DOCS_UTF8_OTHER = \
-	georgian/utf8-doc.tex \
-	german/braille-utf8-doc.tex \
-	greek/utf8-doc.tex \
-	greek/polytonic-utf8-doc.tex \
-	hebrew/utf8-doc.tex \
-	klingon/utf8-doc.tex
-
 DOCS_CPS = \
 	$(DOCS_CP1250) \
 	$(DOCS_CP1252) \
@@ -1194,6 +224,10 @@
 	$(DOCS_LATIN5) \
 	$(DOCS_LATIN9)
 
+DOCS_OTHER = \
+	$(DOCS_ISO88597) \
+	$(DOCS_ISO88598)
+
 DOCS_UTF8 = \
 	$(DOCS_UTF8_LAT) \
 	$(DOCS_UTF8_CYR) \
@@ -1270,7 +304,7 @@
 	alphabets-inc.tex \
 	alphabets-doc.toc
 
-EXTRA_DIST = $(SCRS) test.xdy rom-comma-kludge.sed
+EXTRA_DIST += $(SCRS) test.xdy
 
-CLEANFILES = $(ALPHABETS) $(DOC_CLEAN)
+CLEANFILES += $(nobase_nodist_lang_DATA) $(DOCS) $(DOC_CLEAN)
 
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/lang.defs xindy/make-rules/alphabets/lang.defs
--- xindy.orig/make-rules/alphabets/lang.defs	2010-01-13 14:37:26.000000000 +0100
+++ xindy/make-rules/alphabets/lang.defs	2010-01-10 22:08:23.000000000 +0100
@@ -0,0 +1,45 @@
+albanian UTF8_LAT LATIN1
+belarusian UTF8_CYR CYR_ISO88595 CYR_ISOIR111 CYR_1251
+bulgarian UTF8_CYR CYR_ISO88595 CYR_KOI8_R CYR_1251
+croatian UTF8_LAT CP1250 LATIN2
+czech UTF8_LAT CP1250 LATIN2
+danish UTF8_LAT CP1252 LATIN9
+dutch ij-as-ij UTF8_LAT LATIN1 ij-as-y UTF8_LAT
+english UTF8_LAT CP1252 LATIN9
+esperanto UTF8_LAT LATIN3
+estonian UTF8_LAT CP1252 LATIN9
+finnish UTF8_LAT CP1252 LATIN9
+french UTF8_LAT CP1252 LATIN9
+general UTF8_LAT CP1252 CP850 LATIN9
+georgian UTF8_OTHER
+german braille UTF8_OTHER din5007 UTF8_LAT LATIN1 duden UTF8_LAT LATIN1
+greek UTF8_OTHER ISO88597 polytonic UTF8_OTHER translit UTF8_LAT LATIN4
+gypsy northrussian UTF8_LAT 
+hausa UTF8_LAT
+hebrew UTF8_OTHER ISO88598
+hungarian UTF8_LAT CP1250 LATIN2
+icelandic UTF8_LAT CP1252 LATIN9
+italian UTF8_LAT LATIN1
+klingon UTF8_OTHER
+kurdish bedirxan UTF8_LAT LATIN5 turkish-i UTF8_LAT LATIN5
+latin UTF8_LAT
+latvian UTF8_LAT LATIN4
+lithuanian UTF8_LAT LATIN4
+lower-sorbian UTF8_LAT CP1250 LATIN2
+macedonian UTF8_CYR CYR_ISO88595 CYR_ISOIR111 CYR_1251
+mongolian cyrillic UTF8_CYR cyrillic
+norwegian UTF8_LAT LATIN1
+polish UTF8_LAT CP1250 LATIN2
+portuguese UTF8_LAT LATIN1
+romanian UTF8_LAT LATIN2
+russian UTF8_CYR CYR_ISO88595 CYR_KOI8_R CYR_1251 translit-iso UTF8_LAT
+serbian UTF8_CYR CYR_ISO88595 CYR_ISOIR111 CYR_1251
+slovak large UTF8_LAT CP1250 LATIN2 small UTF8_LAT CP1250 LATIN2
+slovenian UTF8_LAT CP1250 LATIN2
+spanish modern UTF8_LAT LATIN1 traditional UTF8_LAT LATIN1
+swedish UTF8_LAT CP1252 LATIN9
+test1
+turkish UTF8_LAT LATIN3 LATIN5
+upper-sorbian UTF8_LAT CP1250 LATIN2
+ukrainian UTF8_CYR CYR_KOI8_U CYR_1251
+vietnamese UTF8_LAT
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/make-rules/alphabets/mk-lang.sh xindy/make-rules/alphabets/mk-lang.sh
--- xindy.orig/make-rules/alphabets/mk-lang.sh	2010-01-13 14:37:26.000000000 +0100
+++ xindy/make-rules/alphabets/mk-lang.sh	2010-01-13 13:51:57.000000000 +0100
@@ -0,0 +1,213 @@
+#! /bin/sh
+
+# Copyright (C) 2010 Peter Breitenlohner <tex-live@tug.org>.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program in the file COPYING; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+# Boston, MA 02110-1301 USA.
+
+# Generate the Makefile fragments lang/Makefile.am from language
+# definitions in lang.defs containing lines:
+#   lang [[prefix] UTF8_DOC [DOC]...]...
+# where
+#   prefix is an optional filename prefix, without final '-'
+#   UTF8_DOC is a DOCS_UTF8_* variable, without leading 'DOCS_'
+#   DOC is an other DOCS_* variable, without leading 'DOCS_'
+
+CP_DOCS='CP1250 CP1252 CP850'
+LATIN_DOCS='LATIN1 LATIN2 LATIN3 LATIN4 LATIN5 LATIN9'
+CYR_DOCS='CYR_1251 CYR_ISO88595 CYR_ISOIR111 CYR_KOI8_R CYR_KOI8_U'
+ISO_DOCS='ISO88597 ISO88598'
+UTF_DOCS='UTF8_LAT UTF8_CYR UTF8_OTHER'
+
+ALL_DOCS="$CP_DOCS $LATIN_DOCS $CYR_DOCS $ISO_DOCS $UTF_DOCS"
+
+err_exit () {
+	echo "bad argument \"$arg\""
+	exit 1
+}
+
+do_XDY () {
+	XDY="$XDY $name"
+	eval DOCS_$arg="\"\$DOCS_$arg $name\""
+}
+
+do_UTF8 () {
+	PLIN="$PLIN $name"
+	do_XDY
+}
+
+do_CODE () {
+	PL="$PL $name"
+	do_XDY
+}
+
+parse_lang () {
+	for DOC in $ALL_DOCS; do
+		eval DOCS_$DOC=
+	done
+	PLIN= PL= XDY=	
+	lang=$1
+	shift
+	pref=
+	for arg
+	do
+		case $arg in
+			[a-z]*) pref="$arg-";;
+			UTF8_*) name=${pref}utf8; do_UTF8;;
+			[A-Z]*)	case $arg in
+					CP*) val="cp${arg:2}";;
+					ISO*) val="iso${arg:3}";;
+					LATIN*) val="latin${arg:5}";;
+					CYR_1251) val=cp1251;;
+					CYR_ISO88595) val=iso88595;;
+					CYR_ISOIR111) val=isoir111;;
+					CYR_KOI8_R) val=koi8-r;;
+					CYR_KOI8_U) val=koi8-u;;
+					*) err_exit;;
+				esac
+				name="$pref$val"
+				do_CODE
+			;;
+			*) err_exit;;
+		esac
+	done
+}
+
+print_header () {
+	cat <<-_EOF
+	## Makefile fragment for $lang.
+
+	## Copyright (C) 2009, 2010 by Peter Breitenlohner.
+	## Copyright (C) 2004-2005 by Gour.
+	##
+	## This program is free software; you can redistribute it and/or
+	## modify it under the terms of the GNU General Public License as
+	## published by the Free Software Foundation; either version 2 of the
+	## License, or (at your option) any later version.
+	##
+	## This program is distributed in the hope that it will be useful, but
+	## WITHOUT ANY WARRANTY; without even the implied warranty of
+	## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+	## General Public License for more details.
+	##
+	## You should have received a copy of the GNU General Public License
+	## along with this program in the file COPYING; if not, write to the
+	## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+	## Boston, MA 02110-1301 USA.
+
+	subdirs += $lang
+	_EOF
+}
+
+print_xdy () {
+	test $# -eq 0 && return
+	printf '\nnobase_nodist_lang_DATA +='
+	for xdy
+	do
+		printf ' \\\n\t'$lang'/%s-lang.xdy' $xdy
+		printf ' \\\n\t'$lang'/%s-test.xdy' $xdy
+		printf ' \\\n\t'$lang'/%s.xdy' $xdy
+	done
+	printf "\n"
+}
+
+print_docs () {
+	test $# -eq 0 && return
+	printf '\nDOCS_%s +=' $DOC
+	printf ' \\\n\t'$lang'/%s-doc.tex' "$@"
+	printf "\n"
+}
+
+print_extra_hdr () {
+	test -n "$print_extra_hdr_done" && return
+	printf '\nEXTRA_DIST +='
+	print_extra_hdr_done=yes
+}
+
+print_extra () {
+	test $# -eq 0 && return
+	print_extra_hdr
+	printf ' \\\n\t%s' "$@"
+}
+
+print_extra_plin () {
+	test $# -eq 0 && return
+	print_extra_hdr
+	printf ' \\\n\t'$lang'/%s.pl.in' "$@"
+}
+
+print_pl () {
+	test $# -eq 0 && return
+	printf '\ndist_noinst_DATA +='
+	printf ' \\\n\t'$lang'/%s.pl' "$@"
+	printf "\n"
+}
+
+print_clean () {
+	test $# -eq 0 && return
+	printf '\nCLEANFILES +='
+	for clean
+	do
+		case $clean in
+			*utf8) printf ' \\\n\t'$lang'/%s.pl' $clean;;
+		esac
+		printf ' \\\n\t'$lang'/%s.raw' $clean
+	done
+	printf "\n"
+}
+
+make_lang () {
+	print_header
+	print_xdy $XDY
+	for DOC in $ALL_DOCS; do
+		eval print_docs \$DOCS_$DOC
+	done
+	print_extra_hdr_done=
+	case $lang in
+		general) PL="$PL template";;
+		klingon) PL="$PL translit-ascii";;
+		test1) PL="$PL test1";;
+		romanian)
+			cat <<\_EOF
+
+romanian/latin2.pl : romanian/utf8.pl.in
+	sed -e 's/@u{\([^}]*\)}//g;' -f $(srcdir)/rom-comma-kludge.sed <$< | \
+	$(CONVERT_FROM_UTF8)ISO8859-2 >$@
+_EOF
+			print_extra rom-comma-kludge.sed
+		;;
+	esac
+	print_extra_plin $PLIN
+	test -n "$print_extra_hdr_done" && printf '\n'
+	print_pl $PL
+	print_clean $XDY
+	echo
+}
+
+all_langs () {
+	while read line
+	do
+		parse_lang $line
+		mkdir -p $lang
+		make_lang >$lang/Makefile.tmp
+		if diff $lang/Makefile.tmp $lang/Makefile.am >/dev/null 2>&1; then
+			rm -f $lang/Makefile.tmp
+		else
+			mv -f $lang/Makefile.tmp $lang/Makefile.am
+		fi
+	done
+}
+
+all_langs < lang.defs
